nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/processor/SimpleProcessLogger.java

```diff
@@ -448,4 +454,20 @@ public void log(LogLevel level, String msg, Object[] os, Throwable t) {
         }
     }
 
+    private String getCauses(final Throwable throwable) {
```

Comment:
We could leverage Java for more readability to reverse the order of the error messages. Something like this:
```java
        final LinkedList<String> causes = new LinkedList<>();
        for (Throwable t = throwable; t != null; t = t.getCause()) {
            causes.push(t.toString());
        }
        Collections.reverse(causes);
        return causes.stream().collect(Collectors.joining(System.lineSeparator() + CAUSES));
```
Also this implementation adds a new line at the start. Is that intentional?

Reply:
Thanks for the tip! Yes, the new line in the beginning was intentionaly for better readability.